sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -76,15 +76,15 @@ def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
     def on_response(
         self,
         request: PipelineRequest[HTTPRequestType],
-        response: PipelineResponse[HTTPRequestType, HTTPResponseType],
+        response: PipelineResponse[HTTPRequestType, HTTPResponseType],  # type: ignore[override]
     ) -> None:
         super().on_response(request, response)
         if self._enable_diagnostics_logging:
             http_response = response.http_response
             options = response.context.options
             logger = request.context.setdefault("logger", options.pop("logger", self.logger))
             try:
-                logger.info("Elapsed time in seconds: {}".format(time.time() - request.context.get("start_time")))
+                logger.info("Elapsed time in seconds: {}".format(time.time() - request.context["start_time"]))
```

Comment:
Seems like request.context.get("start_time") is better?

Reply:
It would need a default value (e.g. `0`) - and I couldn't think of a good one that makes sense.
Currently if the `get()` returns `None`, then the time calculation will explode with a type error (which is why mypy complained). Now with this change it will fail with a key error instead.
Either way it doesn't make much difference as either error would have been caught by the broad-except below - but this one makes mypy happy :)

Reply:
An alternative would be to do something like:
```python
elapsed: Union[str, float] = time.time() - request.context["start_time"] if 'start_time' in request.context else "<unknown>"
logging.info(f"Elapsed time in seconds: {elapsed}"}
```

Reply:
Given this is logging, I guess we don't want to have uncaught error here if `start_time` is not in request.context?

IOW, maybe we can just give a default value (e.g. `0`)?

Reply:
I would rather replace it with something like "unknown", given that using a default of 0 will show up in the logging as:
`Elapsed time in seconds: 1704924264.5151117`
Which would be nonsensical at best :)

Reply:
I agree on both accounts - we definitely shouldn't be erroring out in the diagnostics, but using a default of 0 wouldn't make sense. Let's set the default to `unknown` like mentioned - that would make sense in letting us know that it wasn't properly captured without the error.

Reply:
Fixed :)